### PR TITLE
apiextensions: prune {any,one}Of + Not recursively on OpenAPI v2 conversion

### PR DESCRIPTION
**Pull request by sttts**
Fixes https://github.com/kubernetes/kubernetes/issues/71132.

### CODE REVIEW

**Review by roycaihw — COMMENTED** [review]
/lgtm

The new unit test capturing the bug has passed.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go [hunk 1]
```diff
@@ -50,8 +50,17 @@ func ValidateCustomResource(customResource interface{}, validator *validate.Sche
...
+// ConvertJSONSchemaProps converts the schema from apiextensions.JSONSchemaPropos to go-openapi/spec.Schema.
 func ConvertJSONSchemaProps(in *apiextensions.JSONSchemaProps, out *spec.Schema) error {
+	return ConvertJSONSchemaPropsWithPostProcess(in, out, nil)
+}
+
+// PostProcessFunc post-processes one node of a spec.Schema.
+type PostProcessFunc func(*spec.Schema) error
+
+// ConvertJSONSchemaPropsWithPostProcess converts the schema from apiextensions.JSONSchemaPropos to go-openapi/spec.Schema
+// and run a post process step on each JSONSchemaProps node.
```
Comment (roycaihw) [inline staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go]: The post process option looks like a good hot-fix to me. It's a little bit too flexible/powerful IMO. 

In future maybe we want a walker processing on a single type like https://github.com/kubernetes/kubernetes/issues/71132 suggested (I checked kube-openapi and it's not straight-forward to use the refWalker there). So we don't have to mix conversion and object-processing up

Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go]: Yes, a walker would be great. This is kind of a walker, but does the conversion in parallel.

File: staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go [hunk 2]
```diff
@@ -86,41 +95,43 @@ func ConvertJSONSchemaProps(in *apiextensions.JSONSchemaProps, out *spec.Schema)
 		out.Example = *(in.Example)
 	}
 
-	out.Enum = make([]interface{}, len(in.Enum))
-	for k, v := range in.Enum {
-		out.Enum[k] = v
+	if in.Enum != nil {
+		out.Enum = make([]interface{}, len(in.Enum))
```
[the remark this answers is not in the document]
Comment (sttts) [reply staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go]: And makes tests green :-)

**Comment by roycaihw** [comment]
/retest

```
[sig-api-machinery] CustomResourceDefinition resources Simple CustomResourceDefinition has OpenAPI spec served with CRD Validation chema
```

[failed](https://gubernator.k8s.io/build/kubernetes-jenkins/pr-logs/pull/71137/pull-kubernetes-e2e-gce/60422) 2 out of 30 times today due to timeout. I think we can increase the timeout to 3 minutes when updating the e2e test

**Pull request merged**